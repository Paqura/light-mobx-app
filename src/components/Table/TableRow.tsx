import React, { MouseEvent } from 'react'

const formatCase = new Intl.NumberFormat().format;

interface Props {
  idx: number;

  data: {
    country: string;
    cases: number;
    countryInfo: { flag: string };
  }

  onChartOpen: (country: string) => (evt: MouseEvent) => void;
}

const TableRow: React.FC<Props> = ({ data, idx, onChartOpen }) => {
  return (
    <tr>
      <td>{idx}</td>
      <td onClick={onChartOpen(data.country)}>
        <button>{data.country}</button>
      </td>
      <td>{formatCase(data.cases)}</td>
      <td>
        <span>
          <img src={data.countryInfo.flag} alt={`Flag of ${data.country}`} width="32px" />
        </span>
      </td>
    </tr>
  )
}

export default TableRow
