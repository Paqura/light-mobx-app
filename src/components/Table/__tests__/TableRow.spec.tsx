import React from 'react';
import TableRow from '../TableRow';

import renderer from 'react-test-renderer';

describe('TableRow', () => {
  it('should return snapshot', () => {
    const wrapper = renderer.create(
      <TableRow
        idx={0}
        key={0}
        onChartOpen={jest.fn()}
        data={{ country: 'USA', cases: 20, deaths: 1, countryInfo: { flag: 'path/to/flag' } }}
      />
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});