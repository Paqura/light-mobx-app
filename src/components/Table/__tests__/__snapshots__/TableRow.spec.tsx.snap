// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow should return snapshot 1`] = `
<tr>
  <td>
    0
  </td>
  <td>
    <button>
      USA
    </button>
  </td>
  <td>
    20
  </td>
  <td>
    1
  </td>
  <td>
    <span>
      <img
        alt="Flag of USA"
        src="path/to/flag"
        width="32px"
      />
    </span>
  </td>
</tr>
`;
